pkg/graph/resolvers/system_intake.go

```diff
@@ -381,15 +383,33 @@ func GetRequesterUpdateEmailData(
 		return nil, err
 	}
 
-	var euaIDs []string
+	var (
+		euaIDs []string
+		euaSet = map[string]struct{}{}
+	)
+
 	for _, item := range data {
+		// de-duplicate
+		if _, seen := euaSet[item.EuaUserID]; seen {
+			continue
+		}
+
+		euaSet[item.EuaUserID] = helpers.EmptyStruct
 		euaIDs = append(euaIDs, item.EuaUserID)
 	}
 
-	// then, get user data from Okta
-	userData, err := fetchUserInfos(ctx, euaIDs)
-	if err != nil {
-		return nil, err
+	// chunk the data into the maximum, which is 200
+	chunks := helpers.Chunk(euaIDs, maxEUAsPerRequest)
+
+	var userData []*models.UserInfo
+	for _, chunk := range chunks {
```

Comment:
it looks like the go slices package has this natively now! Perhaps it would make more sense to use the built in functionality so we don't have to maintain the helper and test code, since GO provides it in the standard library
```suggestion
	for chunk := range slices.Chunk(euaIDs, maxEUAsPerRequest) {
```